nucleus/annotation.py

```diff
@@ -41,6 +42,9 @@
     Z_KEY,
 )
 
+# TODO: refactor to reduce this file to under 1000 lines.
+# pylint: disable=C0302
```

Comment:
Added this exact copy for consistency (used in `dataset.py`), though I'm not sure how much we can really refactor here. Perhaps we can split this file into a directory, where each file is a distinct annotation type.

Reply:
Yeah, we should split this file up, we can then change this into an aggregate file where we just import and expose this as a common interface for the classes.